engine/runtime-compiler/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -133,12 +133,17 @@ class Compiler(
     *
     * @param shouldCompileDependencies whether compilation should also compile
     *                                  the dependencies of the requested package
+    * @param shouldWriteCache whether the compilation results should be written
+    *                         to the cache; if set to False, a 'lint' compilation
+    *                         will be performed, reporting any problems,
+    *                         but no results will be written
     * @param useGlobalCacheLocations whether or not the compilation result should
     *                                  be written to the global cache
     * @return future to track subsequent serialization of the library
     */
   def compile(
     shouldCompileDependencies: Boolean,
+    shouldWriteCache: Boolean,
```

Comment:
I assume this is a reasonable way to avoid the _static type information_ to be stored in the caches.

Reply:
Yes, plus IIRC it fixes a weird behaviour - that even if caches were disabled, they would still be written after compilation or something.

Now, if we run `--no-ir-cache`, we neither _read_ **nor _write_** the caches, which I think is correct.

Reply:
The reason why we are seeing failures like https://github.com/enso-org/enso/actions/runs/12680614808/job/35342759884?pr=12023#step:7:2875 is probably because of this **nor write** behavior. Previously there was one failing run and the bad caches would get overwritten. Now the bad caches stick there forever.

That's rational, but looks like the 

https://github.com/enso-org/enso/blob/cb72487cc962a4f3adc37845639363ecf847f4a4/engine/runtime-benchmarks/src/main/java/org/enso/compiler/benchmarks/module/ImportStandardLibrariesBenchmark.java#L129

may not be ready for that!

Reply:
Oh...

Can't we just run the engine with `--compile` to write the new caches, before running this bench?